hecate_lexer/src/token.rs

```diff
@@ -1,9 +1,32 @@
-use hecate_util::ast::{BinaryOp, UnaryOp};
-
+#[derive(Debug, PartialEq, Eq)]
 pub enum Token {
     Identifier,
-    BinaryOp(BinaryOp),
-    UnaryOp(UnaryOp),
-    AssignmentOp,
+    Operator(Operator),
+    Assignment,
```

Comment:
> Assignment is a confusing operator to have as a standalone token

I will move `Assignment` into the `Operator` enum.

> why AssignmentOp is a special case.

`AssignmentOp` does not exist anymore, I am not quite sure what you mean.

> difficult distinction between `<` (less than) and `<` (generic params begin)

I think the Lexer should not match brackets, it should at most check if the operator is followed by a whitespace. But for this case it would not be optimal either as
```
let var1: i32 = 5;
let var2: i32 = 10;
if var1>var2 { ... }
```
would not be tokenized correctly. So we will indeed have to find a solution to that, as most operators are used for different things in different contexts, but the lexer should not be context aware.